content/blog/15-making-rust-attractive-for-writing-gtk-applications.md

```diff
@@ -0,0 +1,301 @@
++++
+title = "Making Rust attractive for writing GTK applications"
+date = 2022-10-23
+draft = false
+
+[taxonomies]
+categories = ["Rust"]
+tags = ["Rust"]
+
+[extra]
+lang = "en"
+toc = true
+math = false
+mermaid = false
+cc_license = true
++++
+
+[Rust](https://www.rust-lang.org/), the programming language, have been gaining traction across the whole stack. From the pretty recent [upstream kernel support](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=8aebac82933ff1a7c8eede18cab11e1115e2062b), writing games, websites, low level OS components or even desktop applications.
+
+The gtk-rs team have been doing an impressive amount of work during the last few years to make the experience of using `GObject` based libraries from Rust enjoyable by providing high quality bindings around those libraries, generated with [gir](https://github.com/gtk-rs/gir) from the [introspection data](https://gi.readthedocs.io/en/latest/).
+
+Approximately two years ago and few months before the release of [GTK](https://gtk.org/) 4, I decided to take over the maintenance of [gtk4-rs](https://github.com/gtk-rs/gtk4-rs) and push forward the initial work made by Xiang Fan during a Google Summer of Code internship. Nowadays, it is the most used GTK 4 bindings out there with probably more than 100 applications written in it, from simple applications like [Contrast](https://flathub.org/apps/details/org.gnome.design.Contrast) to complex ones like [Telegrand](https://github.com/melix99/telegrand) or [Spot](https://flathub.org/apps/details/dev.alextren.Spot).
```

Comment:
How do you know it's the most used GTK 4 binding? Someone might say this is not true :)

Reply:
I have been following the applications submitted to GNOME Circle the last couple of months and most of the fancy new things are written in Rust in general followed by GJS